src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,397 @@
+        }
+    }
+
+    co_return rpk::debug_bundle_error::success;
+}
+
+rpk::debug_bundle_error detail::make_rpk_err_value(std::error_code err) {
+    return rpk::debug_bundle_error(err.value());
+}
+
+ss::future<std::error_code> debug_bundle::do_remove(ss::sstring& filename) {
+    co_await ss::remove_file(
+      detail::make_bundle_filename(get_write_dir(), filename));
+    co_await ss::sync_directory(get_write_dir().c_str());
+
+    co_return rpk::debug_bundle_error::success;
+}
+
+ss::future<std::error_code> debug_bundle::remove_bundle(ss::sstring& filename) {
+    gate_guard guard{_rpk_gate};
+    auto units = co_await _rpk_mu.get_units();
+
+    auto err = co_await is_ready(filename);
+    if (detail::make_rpk_err_value(err) != rpk::debug_bundle_error::success) {
+        co_return err;
+    }
+
+    co_return co_await container().invoke_on(
+      debug_bundle_shard_id,
+      [&filename](debug_bundle& b) { return b.do_remove(filename); });
+}
+
+ss::future<std::error_code> debug_bundle::do_nonblocking_abort() {
+    gate_guard guard{_rpk_gate};
+
+    if (!_process.has_value()) {
+        co_return rpk::debug_bundle_error::nothing_running;
+    }
+
+    // First try graceful stop
+    _process->terminate();
+    auto wstatus = co_await _process->wait();
+    auto err = detail::handle_wait_status(
+      wstatus, ss::sstring{"Graceful abort"}, default_bundle_name());
+    if (detail::make_rpk_err_value(err) != rpk::debug_bundle_error::success) {
+        // Do forceful stop if graceful failed
+        vlog(
+          bundle_log.debug,
+          "Failed to gracefully abort {}. Running SIGKILL ...",
+          default_bundle_name(),
+          err);
+        _process->kill();
+    }
+
+    co_return rpk::debug_bundle_error::success;
+}
+
+ss::future<std::error_code> debug_bundle::abort_bundle() {
+    auto bundle_status = co_await get_status();
+    if (bundle_status == debug_bundle_status::not_running) {
+        co_return rpk::debug_bundle_error::nothing_running;
+    }
+
+    auto err = co_await container().invoke_on(
+      debug_bundle_shard_id, [](debug_bundle& b) {
+          return b.do_nonblocking_abort().handle_exception_type(
+            [](const std::system_error& ec) {
+                // A system error is thrown when SIGTERM/SIGKILL are called on
+                // an already dead process, handle that case here.
+                vlog(bundle_log.error, "Abort failed, err {}", ec);
+                return ss::make_ready_future<std::error_code>(
+                  rpk::debug_bundle_error::nothing_running);
+            });
+      });
+
+    if (detail::make_rpk_err_value(err) != rpk::debug_bundle_error::success) {
+        co_return err;
+    }
+
+    // Acquire the lock here so we guard against other requests setting the
+    // status or removing the bundle from disk. This is OK since the
+    // non-blocking abort already happenend.
+    gate_guard guard{_rpk_gate};
```

Comment:
I'm a little confused why the gate_guard is down here and not at the start of the function.

I'm a little confused as to what purpose gate_guard is being used for.  (I know the overall purpose of a gate, but it's more a thought of if it's being used properly here).

Reply:
IMO, all requests to the `debug_bundle` should go through the same gate. I agree that it should go at the  beginning of this method, that was a typo.